File: analyze_results.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const logFile = path.join(__dirname, 'logs', 'signals.csv');

if (!fs.existsSync(logFile)) {
    console.log(`File ${logFile} not found.`);
    process.exit(1);
}

const content = fs.readFileSync(logFile, 'utf8');
const lines = content.trim().split('\n');

if (lines.length < 2) {
    console.log("Log file is empty or only header.");
    process.exit(0);
}

const header = lines[0].split(',');
const data = lines.slice(1).map(line => {
    const values = line.split(',');
    // Handle quoted values if any (simple implementation)
    return values;
});

// Indices based on header: timestamp,entry_minute,time_left_min,regime,signal,model_up,...
const idxTimestamp = header.indexOf('timestamp');
const idxSignal = header.indexOf('signal');
const idxRec = header.indexOf('recommendation');
const idxModelUp = header.indexOf('model_up');

const parseDate = (d) => new Date(d);

const timestamps = data.map(r => parseDate(r[idxTimestamp])).filter(d => !isNaN(d));
if (timestamps.length === 0) {
    console.log("No valid timestamps found.");
    process.exit(0);
}

const startTime = new Date(Math.min(...timestamps));
const endTime = new Date(Math.max(...timestamps));
const durationMs = endTime - startTime;
const hours = Math.floor(durationMs / 3600000);
const minutes = Math.floor((durationMs % 3600000) / 60000);

console.log(`Analysis Period: ${startTime.toISOString()} to ${endTime.toISOString()}`);
console.log(`Duration: ${hours}h ${minutes}m`);
console.log(`Total Log Entries: ${data.length}`);

const signals = {};
const recommendations = {};
let sumModelUp = 0;
let countModelUp = 0;

data.forEach(row => {
    const sig = row[idxSignal];
    const rec = row[idxRec];
    const modUp = parseFloat(row[idxModelUp]);

    signals[sig] = (signals[sig] || 0) + 1;
    recommendations[rec] = (recommendations[rec] || 0) + 1;
    
    if (!isNaN(modUp)) {
        sumModelUp += modUp;
        countModelUp++;
    }
});

console.log("\n--- Signals Breakdown ---");
console.table(signals);

console.log("\n--- Recommendations Breakdown ---");
console.table(recommendations);

if (countModelUp > 0) {
    console.log(`\nAverage Model Up Probability: ${(sumModelUp / countModelUp).toFixed(4)}`);
}

File: analyze_results.py
import pandas as pd
import os

log_file = 'logs/signals.csv'

if not os.path.exists(log_file):
    print(f"File {log_file} not found.")
    exit()

try:
    df = pd.read_csv(log_file)
except Exception as e:
    print(f"Error reading CSV: {e}")
    exit()

# Basic stats
if df.empty:
    print("Log file is empty.")
    exit()

df['timestamp'] = pd.to_datetime(df['timestamp'])
start_time = df['timestamp'].min()
end_time = df['timestamp'].max()
duration = end_time - start_time

print(f"Analysis Period: {start_time} to {end_time}")
print(f"Duration: {duration}")
print(f"Total Log Entries: {len(df)}")

print("\n--- Signals Breakdown ---")
print(df['signal'].value_counts())

print("\n--- Recommendations Breakdown ---")
print(df['recommendation'].value_counts())

print("\n--- Average Confidence (Model Up) ---")
print(df['model_up'].mean())

# Filter for actual trades (assuming 'BUY UP' or 'BUY DOWN')
trades = df[df['signal'].isin(['BUY UP', 'BUY DOWN'])]
if not trades.empty:
    print("\n--- Trade Signals ---")
    print(f"Total Trade Signals: {len(trades)}")
    
    # Group by consecutive signals to estimate unique trade opportunities (simple heuristic)
    # If timestamps are close (e.g. < 5 min), count as 1 event?
    # Or just show signal count
    print("First trade signal:", trades['timestamp'].min())
    print("Last trade signal:", trades['timestamp'].max())
else:
    print("\nNo BUY signals found.")
